### ISSUE TITLE
how to abort a request?

### ISSUE THREAD

**Issue by epikhighs** [basti1302/traverson#27]
Hello!

I was curious what's the recommended way of aborting a request?  Both superagent and request libs seem to support it, but I'm not sure if aborting a request is exposed in traverson or how to best expose such a feature.

For my particular use case (not sure if this would simplify implementation), I'd be OK with aborting a request after all the links have been 'followed'.

Thanks.

**Comment by basti1302** [comment]
Currently there is no way to do this. If you have a good idea for API or implementation, let's discuss it :-)

One remark: I think only aborting the last request but let Traverson do all the intermediate requests would feel a bit weird. I think it would feel more natural to abort the whole follow process and current request (if one is in progress at the moment). Reason: Traverson hides individual requests to a certain degree from the client and offers a slighlty higher level API with the follow method - so we might want to offer the ability to abort on that level.

**Comment by epikhighs** [comment]
Thanks for responding so quickly!

Regarding a public API, I would think that returning some object with an abort method from Builder.prototype.get and Builder.prototype.getResource makes sense, since these get() and getResource() is what actually initiates the request.  

```javascript

Builder.prototype.getResource = function(callback) {
...
return {
  abort: function () { ... }
};
};

Builder.prototype.getResource = function(callback) {
...
return {
  abort: function () { ... }
};
};
```
Now, how to implement this, I'd need to do some digging around.  But I would think 2 problems need to be solved:

1. Expose the native request object (from superagent or request lib) to traverson's private/protected methods so that eventually request.abort() can be called.

2. Figure out how to stop subsequent following of links if they are in progress.

**Comment by epikhighs** [comment]
Could the custom request library feature be used somehow to solve this problem?

**Comment by basti1302** [comment]
Working on this now.

What would you expect to happen, if the link traversal process is aborted? Would the callback that was given to `get()`/`getResource()`/`post()`/... still be called? If so, would it be called with an error (something like `new Error('Link traversal process has been aborted.')`)?

**Comment by basti1302** [comment]
Landed in version 1.1.0

**Comment by epikhighs** [comment]
Nice thanks @basti1302 !
